### PR TITLE
chore(all): Android Compile SDK Version updated to 34 for all

### PR DESCRIPTION
**Pull request by Faiizii**
## Description
- Android Compile SKD updated to 34 for all since it is available for developers now

## Related Issues

NA 

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by vbuberen** [comment]
Thanks for your contribution. I would ask you to split bumping compile SDK version and tests refactoring into different PRs to not mix contexts and have proper changelog generated during release.

**Comment by vbuberen** [comment]
Thanks for updating PRs. I would like to merge this one when Android 14 is officially released, so for now it has to wait for a few more weeks till the release.

**Comment by Faiizii** [comment]
Thanks @vbuberen

**Comment by vbuberen** [comment]
As Android 14 is supposed to be released along with Pixel 8 series, going to merge this PR.

**Review by vbuberen — APPROVED** [review]

**Pull request merged**
